### DISCUSSION TITLE
Getting problem in importing Test.sol

### DISCUSSION THREAD

**Discussion by ikunal-04** [Cyfrin/foundry-full-course-cu#680] [Q&A] [answered]
Here's my code:
```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Test, console} from "../forge-std/Test.sol";
import {FundMe} from "../src/FundMe.sol";

contract FundMeTest is Test {
    uint256 no = 1;

    function setUp() external {
        no = 2;
    }

    function testDemo() public {
        console.log(no);
        assertEq(no, 2);
    }
}
```
And this is the error I'm getting when running ``` forge build ``` command
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/34b83a6e-584d-4259-95d7-42f5eaa97f02)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/905fed24-2b6b-4300-86d6-4772dd5fc937)

**Comment by shikhar229169** [comment]
import {Test, console} from "forge-std/Test.sol";

Try with this one

**Reply by ikunal-04** [reply]
I have tried it already, but this is also not working

**Reply by theirrationalone** [reply]
Please Try to update foundry with the known command `foundryup`... and if you get same error in the next run, then Please Follow this [discussion](https://github.com/Cyfrin/foundry-full-course-f23/discussions/644#discussioncomment-6730003) , The fact is... Something went  wrong with your foundry binaries... And Don't forget to restart `vs-code` in both cases.

Just after update: don't forget to update the source... like this...

```terminal
source /Users/<your_username>/.bashrc
```

And Please follow same import technique as directed by @shikhar229169  to you in the previous comment.

And don't forget to verify your `foundry.toml` file -> [profile.default]....

e.g., 
```toml
[profile.default]
src = "src"
out = "out"
libs = ["lib"]    # Make sure to this 😉
```

**Reply by ikunal-04** [reply]
It's still not working I have tried both the ways, followed the same import technique as directed by @shikhar229169 and also verified the foundry.toml file. And also this cmd is giving error:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/232cf998-77a2-48e5-8bf8-801be362a047)

I have tried with both the linux and windows usernames

**Reply by theirrationalone** [reply]
Please remove those `<` & `>` quoted from username. Then try again.... 

E. G., :... 

```terminal
source /Users/theirrationalone/.bashrc
```

**Reply by ikunal-04** [reply]
Still not working I don't know what the problem has occurred but from the start importing test.sol is giving error:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/02960c80-b727-4b82-88cb-5b8272eed2b7)

**Reply by theirrationalone** [reply]
Hey you might ruined whole foundry.... Please try these steps... 

1. Reinstall: foundry.. 

```terminal
curl -L https://foundry.paradigm.xyz | bash
```

2. Update linux/wsl instance.... 

```terminal
sudo apt update
```

3. Update source... 

```terminal
source /Users/theirrationalone/.bashrc
```
4. Try again. 

Please verify this....  If .bashrc exists Or not... As your error said this file doesn't exist. Just focus on that only

![IMG_20230822_115530.jpg](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/f7a930f3-ae52-4ddf-b613-c45efd741fcd)

**Reply by ikunal-04** [reply]
Done, actually this following command worked but still I am receiving the same error when running ```forge test``` or ```forge build```
```
source /home/sallubhai/.bashrc
```

**Reply by theirrationalone** [reply]
Would you like to share that error full screenshot in maximize terminal view??

**Reply by ikunal-04** [reply]
yess sure:
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/f2ce9770-d428-4162-8d32-23f20d41b465)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/1baffe39-986b-45c7-ae92-10e6bc2f2a32)

**Reply by theirrationalone** [reply]
Please put this info in your foundry.toml... 

```toml
remappings = ["ds-test/=lib/forge-std/lib/ds-test/src/", 
 "forge-std/=lib/forge-std/src/"]
```

After doing this if you're still in issue then, Please leave this modification as it is, and make a remappings.txt file parallel to foundry.toml file and fill that file with this data as like this... 

```txt
ds-test/=lib/forge-std/lib/ds-test/src/
forge-std/=lib/forge-std/src/
```

And then try again. Hope this time things would start working 😅

**Reply by ikunal-04** [reply]
Just one doubt I already have a remapping directing to chainlink brownie in foundry.toml so should i put the above command in same bracket?

**Reply by theirrationalone** [reply]
Yes with a comma `, ` like as we do in other langs. 

But don't wrap those paths into an array in `remappings.txt` file. Simply paste as i gave into txt file.

**Reply by ikunal-04** [reply]
okay i have tried both the ways but now error become short but still it is appearing🥲

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/620ec0a2-627d-4d05-aea6-9f863bd3992c)

Yes I did the same as you directed:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/112847288/c628b227-b2fc-41f1-8b7b-c675ddcb9129)

**Reply by theirrationalone** [reply, marked as answer]
You need to install forge-std again. 

Try this... 

```terminal
forge install foundry-rs/forge-std@v1.5.3 --no-commit
```

**Reply by ikunal-04** [reply]
Thank you very much @theirrationalone it worked 🙌

**Reply by theirrationalone** [reply]
**APPRECIATED ❤**

Voila! 😉😅.... Now 10x your coverage on course go...

**Reply by ShawnFromTexas** [reply]
Solved my issue! Thank you was about to take a breakkkkkk!!! :)
